src/tuesday/core/plotting/powerspectra.py

```diff
@@ -0,0 +1,325 @@
+    label : list, optional
+        Label for the PS.
+    smooth : float, optional
+        Standard deviation for Gaussian smoothing.
+        If True, uses a standard deviation of 1.
+    legend_kwargs : dict, optional
+        Keyword arguments for the legend.
+    """
+    rcParams.update({"font.size": fontsize})
+    wavemodes = power_spectrum.k
+    power_spectrum = power_spectrum.ps
+    if color is None:
+        color = "C0"
+    if xlabel is None:
+        xlabel = f"k [{wavemodes.unit:latex_inline}]"
+
+    if ylabel is None:
+        ylabel = f"[{power_spectrum.unit:latex_inline}]"
+        if power_spectrum.unit == un.mK**2:
+            ylabel = r"$\Delta^2_{21} \,$" + ylabel
+        elif power_spectrum.unit == un.dimensionless_unscaled:
+            ylabel = r"$\Delta^2_{21}$"
+        else:
+            ylabel = r"$P(k) \,$" + ylabel
+    if smooth:
+        power_spectrum = gaussian_filter(power_spectrum, sigma=smooth)
+    ax.plot(wavemodes, power_spectrum, color=color, label=label)
+    if title is not None:
+        ax.set_title(title, fontsize=fontsize)
+    ax.set_xlabel(xlabel, fontsize=fontsize)
+    ax.set_ylabel(ylabel, fontsize=fontsize)
+    if log[0]:
+        ax.set_xscale("log")
+    if log[1]:
+        ax.set_yscale("log")
+    if label is not None:
+        ax.legend(**legend_kwargs)
+    return ax
+
+
+def plot_2d_power_spectrum(
+    power_spectrum: CylindricalPS,
+    *,
+    ax: plt.Axes | None = None,
+    title: str | None = None,
+    xlabel: str | None = None,
+    ylabel: str | None = None,
+    clabel: str | None = None,
+    cmap: str | None = "viridis",
+    fontsize: float | None = 16,
+    vmin: float | None = None,
+    vmax: float | None = None,
+    log: list[bool] | None = False,
+    smooth: float | bool = False,
+    cbar: bool | None = True,
+) -> tuple[plt.Figure, plt.Axes]:
+    """
+    Plot a 2D power spectrum.
+
+    Parameters
+    ----------
+    power_spectrum : CylindricalPS
+        Instance of the CylindricalPS class.
+    axs : plt.Axes | list[plt.Axes], optional
+        Axes object(s) to plot on. If None, new axes are created.
+    title : str, optional
+        Title(s) of the plot.
+    xlabel : str, optional
+        Label for the x-axis.
+    ylabel : str, optional
+        Label for the y-axis.
+    clabel : str, optional
+        Label for the colorbar.
+    cmap : str, optional
+        Colormap for the plot.
+    fontsize : float, optional
+        Font size for the plot labels.
+    vmin : float, optional
+        Minimum value for the color scale.
+    vmax : float, optional
+        Maximum value for the color scale.
+    log : list[bool], optional
+        List of booleans to set the kperp, kpar, and PS axes to log scale.
+    smooth : float, optional
+        Standard deviation for Gaussian smoothing.
+        Default is False, if True, uses a standard deviation of 1.
+    """
+    rcParams.update({"font.size": fontsize})
+    kperp = power_spectrum.kperp
```

Comment:
I suggest having an `if not isinstance(power_spectrum, CylindricalPS): raise TypeError()` with a nice message